projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -186,6 +186,14 @@ private[projection] class R2dbcOffsetStore(
     (projection_name, projection_key, slice, persistence_id, seq_nr, timestamp_offset, timestamp_consumed)
     VALUES (?,?,?,?,?,?, transaction_timestamp())"""
 
+  private def insertTimestampOffsetBatchSql(pid: Pid, seqNr: SeqNr, offsetTimestamp: Instant): String = {
+    val slice = persistenceExt.sliceForPersistenceId(pid)
+    sql"""
+      INSERT INTO $timestampOffsetTable
+      (projection_name, projection_key, slice, persistence_id, seq_nr, timestamp_offset, timestamp_consumed)
+      VALUES ('${projectionId.name}','${projectionId.key}',$slice,'$pid',$seqNr,'$offsetTimestamp', transaction_timestamp())"""
```

Comment:
Isn't this risky, `$pid` could be provided by a user and contain `'` or some other sneakery. Or is it escaped elsewhere?

Reply:
yes, sql injection is a major reason for using bind parameters. Maybe we could have some basic checks like not allowing `'` in the string values.

Reply:
Double checked the proxy in case we do some escaping/sanitation there but there's nothing like that, we accept whatever string we get.